instrumentor/controllers/deleteinstrumentationconfig/namespace_controller.go

```diff
@@ -165,11 +169,24 @@ func syncGenericWorkloadListToNs(ctx context.Context, c client.Client, kind work
 // when reconciling the namespace, the usecase is to delete instrumentation for workloads that were only
 // instrumented due to the label on the namespace. These are workloads with the label missing.
 // (they inherit the instrumentation from the namespace this way)
-func isInheritingInstrumentationFromNs(obj client.Object) bool {
+func isInheritingInstrumentationFromNs(ctx context.Context, c client.Client, obj client.Object) (bool, error) {
+	sourceList, err := v1alpha1.GetWorkloadSources(ctx, c, obj)
+	if err != nil {
+		return false, err
+	}
+
+	if sourceList.Namespace != nil && sourceList.Namespace.DeletionTimestamp.IsZero() {
```

Comment:
WDYT about adding a function to utils that gets a source and returns if it's being deleted or not. could be more fluent to read, e.g.

```suggestion
	if sourceList.Namespace != nil && !k8sutils.IsTerminating(sourceList.Namespace) {
```

or:
```suggestion
	if sourceList.Namespace != nil && k8sutils.IsActive(sourceList.Namespace) {
```

We use it in many places and the double negative can be confusing to read :)

The utils function can also check the `nil` and make usage even shorter and more easy to understand.

Reply:
good idea, done